src/windshaft/named-map.js

```diff
@@ -13,13 +14,18 @@ var NamedMap = MapBase.extend({
     //       "2": "/** torque visualization */\n\nMap { ... }"
     //     }
     //   }
-    //
-    json.styles = this._layersCollection.reduce(function (p, c, i) {
-      var style = c.get('cartocss');
+
+    var layers = this._layersCollection.filter(function (layer) {
+      return layer.get('type') !== 'GMapsBase';
```

Comment:
I don't think this will work, the point is the index should change when a map has a google basemaps. A style's layer with index 1 must end having an index 0 in the final object. We could create a new indexish variable to take the count, but I feel more complicated.